server/functions/index.js

```diff
@@ -263,25 +303,33 @@ exports.expireOldFavorsOnCreate = functions.firestore
 // send new favor notification to users on updates
 exports.sendNotificationOnNewChat = functions.firestore
     .document('/chats/{chatId}')
-    .onCreate((snap, context) => {
-        const isFirstMsg = snap.data().isFirstMsg;
-        if (isFirstMsg === "true") {
-            const titleToSend = "You've got a new message";
-            const favorId = snap.data().favorId;
-            const receivers = [snap.data().notifId];
-            const message = {
-                data: {
-                    FavorId: favorId,
-                },
-                notification: {
-                    title: titleToSend,
-                    body: 'Click to check out the chat message',
-                },
-                tokens: receivers
-            };
-            return sendMulticastMessage(message, receivers);
+    .onCreate((snap) => {
+        const message = snap.data();
+        const senderId = message.uid;
+        const favorId = message.favorId;
 
-        }
+        // find favor with selected favorId
+        return db.collection('/favors').where('id', '==', favorId).get()
+            .then((snapshot) => {
+                snapshot.forEach((doc) => {
+                    var favor = doc.data();
+                    var userIds = favor.userIds;
+
+                    var notification = {
+                        data: {
+                            FavorId: favorId,
+                        },
+                        notification: {
+                            title: 'New message in the chat of favor ' + favor.title,
+                            body: 'Click to check out the post details',
+                        },
+                        tokens: []
+                    };
+
+                    userIds.splice(userIds.indexOf(senderId), 1)
+                    sendMessageToUsers(notification, userIds, CHAT_TYPE)
```

Comment:
So right now we send notification to only users who have commited right?

Reply:
We send chat notifications to all users who committed and the requester when no user has been accepted. We then send notification to only the accepter and the requester when a user is accepted.

Reply:
ok great.